regression/threaded_data_transfer/regression_large_data_async_transfers.cpp

```diff
@@ -0,0 +1,126 @@
+//this tests async data transfer between the host and the device using 4 host arrays and 1 device array
```

Comment:
This C++ test is identical to the above C test; do we need both C and C++ files? (all the C and C++ file pairs in this directory are identical to each other) 
If the purpose of the C++ file is to test the C++ compiler, we can compile the same C file using the C++ compiler.